storage/reads/array_cursor.gen.go.tmpl

```diff
@@ -112,6 +112,82 @@ LOOP:
 	return c.res
 }
 
+type {{.name}}ArrayCursor struct {
+	cursors.{{.Name}}ArrayCursor
+	cursorContext
+	filter *{{$type}}
+}
+
+
+func (c *{{.name}}ArrayCursor) reset(cur cursors.{{.Name}}ArrayCursor, cursorIterators cursors.CursorIterators, cond expression) {
+	if cond != nil {
+		if c.filter == nil {
+			c.filter = new{{.Name}}FilterArrayCursor(cond)
+		}
+		c.filter.reset(cur)
+		cur = c.filter
+	}
+
+	c.{{.Name}}ArrayCursor = cur
+	c.itrs = cursorIterators
+	c.err = nil
+}
+
+
+func (c *{{.name}}ArrayCursor) Err() error { return c.err }
+
+func (c *{{.name}}ArrayCursor) Stats() cursors.CursorStats {
+	return c.{{.Name}}ArrayCursor.Stats()
+}
+
+func (c *{{.name}}ArrayCursor) Next() {{$arrayType}} {
+	for {
+		a := c.{{.Name}}ArrayCursor.Next()
+		if a.Len() == 0 {
+			if c.nextArrayCursor() {
+				continue
+			}
+		}
+		return a
+	}
+}
+
+func (c *{{.name}}ArrayCursor) nextArrayCursor() bool {
+	if len(c.itrs) < 1 {
+		return false
+	}
+
+	c.{{.Name}}ArrayCursor.Close()
+
+	var itr cursors.CursorIterator
+	var cur cursors.Cursor
+    itr, c.itrs = c.itrs[0], c.itrs[1:]
```

Comment:
OSS uses cursor iterator lists. In order to follow that convention here, I have to access the list to get the cursorIterator value.